extensions/s3/src/main/java/io/deephaven/extensions/s3/S3ChannelContext.java

```diff
@@ -242,39 +376,50 @@ boolean isDone() {
             return consumerFuture.isDone();
         }
 
+        @Override
+        public ByteBuffer get() {
+            throw new UnsupportedOperationException("Use acquire() and release() to manage the request");
```

Comment:
You are right, the idea here was to not expose the get() method. I can make soft reference a field inside the Request class rather than extending from soft reference.

Reply:
I think if we go with a reachability-driven approach, we no longer want to UOE from `get()`; "get" becomes "acquire()" semantically. That said, we still never want to touch the buffer directly, since we want to work through `fill`.
I'm not super concerned with exposing `get()` in the interface; it's an implementation detail, sure, but it's an efficiency, and we're not really writing this as a utility for the masses.